metrics/prometheus.go

```diff
@@ -571,3 +581,11 @@ func specMemoryValue(v uint64) float64 {
 	}
 	return float64(v)
 }
+
+var invalidLabelCharRE = regexp.MustCompile(`[^a-zA-Z0-9_]`)
+
+// sanitizeLabelName replaces anything that doesn't match
+// client_label.LabelNameRE with an underscore.
+func sanitizeLabelName(name string) string {
```

Comment:
```
Label names may contain ASCII letters, numbers, as well as underscores.
```

From: http://prometheus.io/docs/concepts/data_model/#metric-names-and-labels
I *think* this makes sense as the default. The other option is to ignore the invalid labels.